Source/CarthageKit/VersionFile.swift

```diff
@@ -237,6 +237,43 @@ struct VersionFile: Codable {
 	}
 }
 
+/// Creates a version file for the current project in the
+/// Carthage/Build directory which associates its commitish with
+/// the hashes (e.g. SHA256) of the built frameworks for each platform
+/// in order to allow those frameworks to be skipped in future builds.
+///
+/// Derives the current project name from `git remote get-url origin`
+///
+/// Returns a signal that succeeds once the file has been created.
+public func createVersionFileForCurrentProject(
+	platforms: Set<Platform>,
+	buildProducts: [URL],
+	rootDirectoryURL: URL
+) -> SignalProducer<(), CarthageError> {
+	let currentProjectAsDependecy = launchGitTask(["remote", "get-url", "origin"])
+		.map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
+		.map { Dependency.git(GitURL($0)) }
```

Comment:
There's not always a remote named `origin`, so this should probably catch errors and default to another value.

Reply:
ok so I will go the full way.

Reply:
Otherwise you could fallback to a string like "CurrentProject" or something